pyproject.toml

```diff
@@ -111,3 +96,19 @@ commands =
     pre-commit: pre-commit run --all-files {posargs}
     docs: sphinx-{posargs:build -EW --keep-going} -T docs build/html
 """
+
+
+[tool.ruff]
+src = ["src", "tests"]
+ignore = [
+    "C408", # Unnecessary collection call - e.g. list(...) instead of [...]
```

Comment:
Ok, did you have to add these 2 ignores because of issues in python3-pip-skeleton, or issues in a project that adopts it?

Reply:
For `C408`:

```
python3-pip-skeleton]$ ruff .
.github/pages/make_switcher.py:62:9: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:101:23: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:104:27: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:151:22: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:152:10: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:158:9: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:164:14: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:171:9: C408 [*] Unnecessary `dict` call (rewrite as a literal)
docs/conf.py:179:16: C408 [*] Unnecessary `dict` call (rewrite as a literal)
```

I think flake8 used to ignore these files, but I think `ruff` is correct to check them - they're `.py` files in this codebase after all. It's also a flag that we had in hyperion generally. If you prefer I could fix these few warnings and remove the ignore here?

---

For `E501` it's again something we had in our hyperion flake8 config, I guess to prevent `black` and `flake8` fighting each other in some oddball case maybe? If `black` is doing line length formatting I don't see any benefit in checking it in `ruff`, but can be persuaded to drop this from the skeleton config, it doesn't actually cause any errors in this repo.

Reply:
`black` used to not touch docstrings and other comments. Is that still the case with the latest versions? If so, you'll need to remove `E501` as an exception.

This page seems to suggest the same: https://beta.ruff.rs/docs/faq/

Reply:
ok, have removed E501 here, if a project using the template needs it they can always re add it.

Reply:
My opinion is that the C408 instances should also be fixed in `conf.py`, rather than adding an exception.

Even if there is a very good reason for using it in conf.py (!?) we should add an exception for this file, not refuse to check elsewhere.